### PR TITLE
setup unauthorized flow

### PR DESCRIPTION
**Pull request by ryanlntn**
Adds unauthorized flow to native share.

https://github.com/infinitered/ExpensifyApp/assets/1944151/b33cf78f-a692-4aab-b1aa-c8bbd45f9d47

Trello: https://trello.com/c/RjO2OWIJ/169-unauthenticated-user-should-be-shown-an-error-and-linked-back-to-the-app-to-sign-in

Note this builds on the patch I did to expose the extension context [here](https://github.com/Expensify/App/commit/0c7d1dcf5dad17ffbd18e06356a8bb959078558a)

### CODE REVIEW

File: ios/share/Info.plist [hunk 1]
```diff
@@ -18,7 +18,7 @@
 	<key>HostAppBundleIdentifier</key>
 	<string>com.chat.expensify.chat</string>
 	<key>HostAppURLScheme</key>
-	<string>new-expensify</string>
+	<string>new-expensify://</string>
```
Comment (lindboe) [inline ios/share/Info.plist:21]: Was this change necessary? I thought I remembered it working without that

File: patches/react-native-share-menu+6.0.0.patch [hunk 2]
```diff
@@ -112,7 +112,16 @@ index e290cce..1c114ff 100644
          if error != nil {
              let exception = NSError(
                  domain: Bundle.main.bundleIdentifier!,
-@@ -130,7 +137,7 @@ public class ShareMenuReactView: NSObject {
+@@ -55,6 +62,8 @@ public class ShareMenuReactView: NSObject {
+         }
+ 
+         viewDelegate.openApp()
++        
++        ShareMenuReactView.detachViewDelegate()
```
Comment (lindboe) [inline patches/react-native-share-menu+6.0.0.patch:120]: I thought we already had this in native-share, did something change?

Ah yeah, found it here, looks like different timing for the two use cases: https://github.com/infinitered/ExpensifyApp/pull/28/files#diff-1aa046d161a1081bd541198ac073c301a3773ce2f3591f5f88a5a6fc85c02e36R110

**Review by lindboe — APPROVED** [review]
Looking good and works for me, just left a question about the plist value for the url scheme

File: ios/share/Info.plist (hunk 1 above)
Comment (ryanlntn) [reply ios/share/Info.plist:21]: Yeah this is super confusing. This one needs the `://`. In the [README](https://github.com/Expensify/react-native-share-menu/blob/master/IOS_INSTRUCTIONS.md#create-app-group-and-url-scheme) they explicitly mention not including it for `CFBundleURLSchemes` but including it for `HostAppURLScheme`. Seems like it wouldn't be too hard to update the code to accept both. I'm also curious if there's a way to get the host app url scheme from native avoiding this altogether.

File: patches/react-native-share-menu+6.0.0.patch (hunk 2 above)
Comment (ryanlntn) [reply patches/react-native-share-menu+6.0.0.patch:120]: Yeah from that issue you sent me the other day I think we might just be able to change the lifecycle method where that is currently getting called. Using `viewWillDisappear` instead [here](https://github.com/Expensify/react-native-share-menu/blob/master/ios/ReactShareViewController.swift#L52C17-L52C33) might be all we need.
